packages/common-components/src/Toasts/ToastContext.tsx

```diff
@@ -0,0 +1,196 @@
+    topRightAppearBox: {
+      animation: `$slideLeft ${animation.translate}ms`,
+      [breakpoints.down("sm")]: {
+        animation: `$slideDown ${animation.translate}ms`
+      }
+    },
+    topLeftContainer: {
+      top: 0,
+      left: 0
+    },
+    topLeftAppearBox: {
+      animation: `$slideRight ${animation.translate}ms`,
+      [breakpoints.down("sm")]: {
+        animation: `$slideDown ${animation.translate}ms`
+      }
+    },
+    bottomRightContainer: {
+      bottom: 0,
+      right: 0
+    },
+    bottomRightAppearBox: {
+      animation: `$slideLeft ${animation.translate}ms`,
+      [breakpoints.down("sm")]: {
+        animation: `$slideUp ${animation.translate}ms`
+      }
+    },
+    bottomLeftContainer: {
+      bottom: 0,
+      left: 0
+    },
+    bottomLeftAppearBox: {
+      animation: `$slideRight ${animation.translate}ms`,
+      [breakpoints.down("sm")]: {
+        animation: `$slideUp ${animation.translate}ms`
+      }
+    },
+    "@keyframes slideRight": {
+      from: { transform: "translate(-100%, 0)" },
+      to: { transform: "translate(0, 0)" }
+    },
+    "@keyframes slideLeft": {
+      from: { transform: "translate(100%, 0)" },
+      to: { transform: "translate(0, 0)" }
+    },
+    "@keyframes slideUp": {
+      from: { transform: "translate(0, 100%)" },
+      to: { transform: "translate(0, 0)" }
+    },
+    "@keyframes slideDown": {
+      from: { transform: "translate(0, -100%)" },
+      to: { transform: "translate(0, 0)" }
+    }
+  })
+})
+
+type ToastContextProps = {
+  autoDismiss?: boolean
+  dismissTimeout?: number
+  defaultPosition?: ToastPosition
+  children: React.ReactNode | React.ReactNode[]
+}
+
+interface ToastContext {
+  addToast: (toastParams: ToastParams) => string
+  updateToast: (toastId: string, toastParams: ToastParams, startDismissal?: boolean) => void
+  removeToast: (toastId: string) => void
+  toasts: Toast[]
+}
+
+const ToastContext = React.createContext<ToastContext | undefined>(
+  undefined
+)
+
+const ToastProvider = ({
+  children,
+  autoDismiss = true,
+  defaultPosition = "topRight",
+  dismissTimeout = 5000
+}: ToastContextProps) => {
+  const classes = useStyles()
+  const [toasts, setToasts] = useState<Toast[]>([])
+
+  const removeToast = useCallback((toastId: string) => {
+    setToasts(toasts.filter((toast) => toast.id !== toastId))
+  }, [toasts])
+
+  const addToast = useCallback((toastParams: ToastParams) => {
+    const id = uuidv4()
+    setToasts((toasts) => ([
+      ...toasts,
+      { id,
+        ...toastParams,
+        toastPosition: toastParams.toastPosition || defaultPosition
+      }
+    ]))
+
+    const isProgressToast = toastParams.progress !== undefined
+    const shouldDismiss = toastParams.autoDismiss || autoDismiss
+    const dismissTimeOut = toastParams.dismissTimeout || dismissTimeout
+
+    if (shouldDismiss && !isProgressToast) {
+      setTimeout(() => {
+        setToasts((toasts) => toasts.filter((toast) => toast.id !== id))
```

Comment:
I also replaced this one, and tested thoroughly, looking ok too. I'll check on brave.
```suggestion
        removeToast(id)
```

Reply:
Tried with brave, and looks good.

Reply:
There is one more thing you can try to see the weird behaviour!
There is a `Toasts Demo` under toasts in storybook, try opening a number of toasts there